pic-sure-api-war/src/main/java/edu/harvard/dbmi/avillach/service/ConfigurationService.java

```
@@ -0,0 +1,118 @@
+package edu.harvard.dbmi.avillach.service;
+
+import edu.harvard.dbmi.avillach.data.entity.Configuration;
+import edu.harvard.dbmi.avillach.data.repository.ConfigurationRepository;
+import edu.harvard.dbmi.avillach.data.request.ConfigurationRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+public class ConfigurationService {
+    private final Logger logger = LoggerFactory.getLogger(ConfigurationRepository.class);
+
+    @Inject
+    ConfigurationRepository configurationRepository;
+
+    // If the request has no uuid it's new so match any uuid with this name &/or kind
+    // else, match any uuid with this name &/or kind that isn't the one we're updating
+    private boolean nameKindPairExists(Configuration config) {
+        Map<String, Object> columns = new HashMap<>();
+        columns.put("name", config.getName());
+        columns.put("kind", config.getKind());
+        return configurationRepository.getByColumns(configurationRepository.query(), columns).stream().map(Configuration::getUuid)
+            .anyMatch(uuid -> config.getUuid() == null || !uuid.equals(config.getUuid()));
+    }
+
+    public Optional<List<Configuration>> getConfigurations(String kind) {
+        try {
+            List<Configuration> configs = kind != null ? configurationRepository.getByColumn("kind", kind) : configurationRepository.list();
+            return Optional.ofNullable(configs);
+        } catch (Exception exception) {
+            logger.error("Error retrieving configurations " + (kind != null ? "with kind " + kind : ""), exception);
+            return Optional.empty();
```

Should we return the actual error?